xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -440,3 +419,318 @@ bool CEnumeratorHD::IsSDRSupported()
+  {
+    conversionsString.append("\n");
+    conversionsString.append(StringUtils::Format(
+        "{} {} / {}{} {:<{}} to {} {:<{}} / {}{} {:<{}}", "*",
+        DX::DXGIFormatToString(c.m_inputFormat), (c.m_inputCS == heuristicsInputCS) ? "*" : " ",
+        (c.m_inputCS == inputNativeCS) ? "N" : " ", DX::DXGIColorSpaceTypeToString(c.m_inputCS), 32,
+        (c.m_outputFormat == outputFormat) ? "*" : " ", DX::DXGIFormatToString(c.m_outputFormat),
+        26, (c.m_outputCS == heuristicsOutputCS) ? "*" : " ",
+        (backbufferColorSpaces.find(c.m_outputCS) != backbufferColorSpaces.end()) ? "bb" : "  ",
+        DX::DXGIColorSpaceTypeToString(c.m_outputCS), 32));
+  }
+
+  CLog::LogFC(LOGDEBUG, LOGVIDEO,
+              "Supported conversions from format {}\n(*: values picked by "
+              "heuristics, N native input color space, bb supported as swap chain backbuffer){}",
+              DX::DXGIFormatToString(inputFormat), conversionsString);
+}
+
+bool CEnumeratorHD::IsFormatSupportedInput(DXGI_FORMAT format)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+  return IsFormatSupportedInternal(format, D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_INPUT);
+}
+bool CEnumeratorHD::IsFormatSupportedOutput(DXGI_FORMAT format)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+  return IsFormatSupportedInternal(format, D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_OUTPUT);
+}
+
+bool CEnumeratorHD::IsFormatSupportedInternal(DXGI_FORMAT format,
+                                              D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT support) const
+{
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return false;
+
+  UINT uiFlags;
+  if (S_OK == m_pEnumerator->CheckVideoProcessorFormat(format, &uiFlags))
+  {
+    if (uiFlags & support)
+      return true;
+  }
+
+  CLog::LogF(LOGERROR, "unsupported format {} for {}.", DX::DXGIFormatToString(format),
+             DX::D3D11VideoProcessorFormatSupportToString(support));
+  return false;
+}
+
+ComPtr<ID3D11VideoProcessor> CEnumeratorHD::CreateVideoProcessor(UINT RateConversionIndex)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return {};
+
+  ComPtr<ID3D11VideoProcessor> videoProcessor;
+
+  // create processor
+  HRESULT hr = m_pVideoDevice->CreateVideoProcessor(m_pEnumerator.Get(), RateConversionIndex,
+                                                    videoProcessor.ReleaseAndGetAddressOf());
+  if (FAILED(hr))
+  {
+    CLog::LogF(LOGDEBUG, "failed creating video processor with error {}.",
+               DX::GetErrorDescription(hr));
+    return {};
+  }
+
+  return videoProcessor;
+}
+
+ComPtr<ID3D11VideoProcessorInputView> CEnumeratorHD::CreateVideoProcessorInputView(
+    ID3D11Resource* pResource, const D3D11_VIDEO_PROCESSOR_INPUT_VIEW_DESC* pDesc)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return nullptr;
```

Comment:
In this one is also a `ComPtr` and is returned nullptr. Probably returning `{}` is equivalent (initializes to NULL ) but for consistence all should be the same.

Reply:
yes, that was c/p :-) I'll make them the same